src/app/profile/profile.page.ts

```diff
@@ -26,40 +26,72 @@ export class ProfilePage {
   async loadFormAndData() {
     await this.loader.showLoading("Please wait while loading...");
     this.profileService.getFormJsonAndData()
-    .pipe(
-      catchError((err) => {
-        this.toastService.presentToast(err?.error?.message || 'Error loading profile data. Please try again later.', 'danger');
-        throw err;
-      }),
-      finalize(async () => await this.loader.dismissLoading())
-    )
-    .subscribe(([formJsonRes, profileFormDataRes]: any) => {
-      if(formJsonRes?.status === 200 || profileFormDataRes?.status === 200){
-        this.formJson = formJsonRes?.result?.data;
-        this.formData = profileFormDataRes?.result;
-        this.mapProfileDataToFormJson(this.formData);
-      }else{
-        this.toastService.presentToast('Failed to load profile data. Please try again later.', 'danger');
-      }
-    });
+      .pipe(
+        catchError((err) => {
+          this.toastService.presentToast(err?.error?.message || 'Error loading profile data. Please try again later.', 'danger');
+          throw err;
+        }),
+        finalize(async () => await this.loader.dismissLoading())
+      )
+      .subscribe(([formJsonRes, profileFormDataRes]: any) => {
+        if (formJsonRes?.status === 200 || profileFormDataRes?.status === 200) {
+          this.formJson = formJsonRes?.result?.data;
+          this.formData = profileFormDataRes?.result;
+          this.mapProfileDataToFormJson(this.formData);
+        } else {
+          this.toastService.presentToast('Failed to load profile data. Please try again later.', 'danger');
+        }
+      });
   }
 
   mapProfileDataToFormJson(formData?: any) {
     this.formJson.image = this.formData.image;
     this.formJson.isUploaded = true;
-    Object.entries(formData).map(([key, value]: any) => {
+
+    this.formJson.forEach((control: any) => {
+      if (!(control.name in formData)) {
+        formData[control.name] = control.defaultValue || '';
+      }
+    });
+
+    if (formData.user_roles) {
+      formData.user_roles = formData.user_roles.map((role: any) => ({
+        value: role.title
+      }));
+    }
+    this.formData.roles = formData.user_roles;
+
+    Object.entries(formData).forEach(([key, value]: [string, any]) => {
       const control = this.formJson.find((control: any) => control.name === key);
+
       if (control) {
-        control.disabled =true;
-        if (control.type === 'select') {
-          control.type = 'text';
-        }else if(control.type === 'checkbox'){
-          control.type = 'chip'
-        }else{
-          control.type = 'text';
+        switch (control.type) {
+          case 'select':
+            control.type = "text";
+            control.value = Array.isArray(value)
```

Comment:
in each case see if the logic can be extracted to a single function if not then try different function, it'll be easier to understand the code